src/app/views/example/overview/table/table.component.ts

```diff
@@ -17,13 +17,16 @@ export class TableComponent implements OnInit {
     public lang: string;
 
     @Input()
-    public filter: TerraFilter<any>;
+    public filter!: TerraFilter<any>;
```

Comment:
Hier ist das mit dem `!` tatsächlich gecheatet^^
Das ist ja ein Input. Und der muss nicht zwangsläufig gesetzt sein. Das ViewChild hat die Component ja selbst in der Hand.
Bei filter solltest du es daher weglassen und unten dann einfach folgendes machen:
```
this._dataSource.filter = this.filter ?? new TerraFilter<any>();
```

Reply:
Verstehe. Dann schmeißt er : 

`table.component.ts:20:12 - error TS2564: Property 'filter' has no initializer and is not definitely assigned in the constructor.
`

Reply:
Okay wenn ich den Typ von Filter `TerraFilter<any> | undefined;` gebe, läut es